### PR TITLE
Combat Robots Race

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
update:
Repair changes:
welder and cable now use the radial menu to repair, and will loop healing 15 damage until its fullly repaired
1 second per loop if someone else heals you 3 if you heal yourself

todo

- [x] finish bugfixing
- [x] look at issue with pref menu preview being black -> hardcode, wontfix in this pr
- [x] essential kits/armor

Adds combat robots roundstart race to the game
image
They can only use their snowflake armor and uniform

+takes 30% less brute (removed)
+cant be set on fire
+no chem processing
+no blood

~metal limbs
~their punches do 10 damage
~head can be reattached
~robot voice style in chat
~3 kinds of chonker armor, comes map-varianted

-takes 30% more damage from all burn sources (removed)
-cant wear most non snowflake armor
-no sprint
-no chems

Naming scheme:
verb+er -number

Pathfinder-123
Digger-123
Miner-123
etc
## Why It's Good For The Game

Muh content
## Changelog
:cl:
add: Nanotrasen is now pleased to say that incompetent "human" marines are being replaced with a new limited production Autonomous Combat Robot model!
add: Robots take more burn damage, less brute damage, process no chems, cant run, can have their head reattached, cant wear anthing but their snowflake armor, have no blood(duh), cant be set on fire
qol: Drastically improved welder and cable repair QOL
imageadd: Sprites by Kuro
soundadd: Sounds provided by slayer
/:cl:

### CODE REVIEW

**Review by Kuro020 — APPROVED** [review]

**Comment by token-map** [comment]
you write that they can reattach their head twice

**Comment by Dolth** [comment]
How would they be repaired?

**Comment by TiviPlus** [comment]
same as synths

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -665,3 +665,31 @@ obj/item/clothing/head/helmet/marine/pilot/green
...
+	desc = "Medium armor plating designed for self mounting on the upper half of TerraGov combat robotics. It has self-sealing bolts for mounting on robotic owners inside."
+	icon_state = "robot_helmet_medium"
+	item_state = "robot_helmet_medium"
+	species_exception = list(/datum/species/robot)
+	flags_item_map_variant = (ITEM_JUNGLE_VARIANT|ITEM_ICE_VARIANT|ITEM_PRISON_VARIANT)
+	soft_armor = list("melee" = 15, "bullet" = 15, "laser" = 15, "energy" = 15, "bomb" = 15, "bio" = 15, "rad" = 15, "fire" = 15, "acid" = 15)
+
+/obj/item/clothing/head/helmet/marine/robot/mob_can_equip(mob/M, slot, warning, override_nodrop)
+	. = ..()
+	if(!isrobot(M))
```
Comment (BraveMole) [inline code/modules/clothing/head/helmet.dm:681]: is it worth making this another var on the item, like species_requirement = list(/datum/species/robot) ?

**Review by BraveMole — CHANGES_REQUESTED** [review]
Also, i assume this is repaired like a synth since the LIMB_ROBOT and ROBOTIC_LIMB?

You'll need to improve synth repairs on another pr then, giving it feedback and possible a radial limb selection.

Giving the special (nanobot or whatever) tool to engineers in their points so they can fix robots coud be interesting too

File: code/modules/clothing/head/helmet.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/clothing/head/helmet.dm:681]: maybe in future but for now its only 6 items and i dont want to bloat this pr

**Review by BraveMole — APPROVED** [review]

**Comment by LoganKele** [comment]
The only issue I see with this is that the xenos will salt...a lot.

**Review by BraveMole — CHANGES_REQUESTED** [review]
Alex said robots were bodyblocking xenos (which makes little sense), can you check on that?

**Comment by joniroq** [comment]
Please for the love of Rouny, change the emote sounds. They are ear-splittingly loud in comparison to everything else in the game

**Comment by MiniMeatwad** [comment]
Can't see their health as xeno

**Comment by MiniMeatwad** [comment]
Also comically overpowered, shouldn't be a roundstart thing

**Comment by Tennessee116** [comment]
>Nanotra**n**sen

**Comment by Sirsnek15** [comment]
I played a round with these, super cool, lots of issues with things like the **map only showing other combat bots and round vatborns,** also getting **no radio from the GIMME closet**, and also weren't allowed by design to use robot talk channel, which is silly in terms of roleplay IMO

**Comment by AndroidMK2** [comment]
Sounds extremely epic, add imdietaly or riot shall start.

**Comment by InterroLouis** [comment]
Why is the helmet so ridiculously low in armor stats?

**Comment by SrSC0ndition** [comment]
They really need to be able to wear boots, no matter what armour you take, there is no difference since you're slow as a turtle

**Comment by AndroidMK2** [comment]
Just played as robot and I gotta say it needs buffs to it's armor and it should be able to wear boots and gloves

**Comment by Al-1ce** [comment]
>and also weren't allowed by design to use robot talk channel, which is silly in terms of roleplay IMO
>TGMC
>Roleplay?


Don't be silly

**Comment by AndroidMK2** [comment]
I really love combat robots but right now they are worse from marines in every way, their 30% brute damage resistance doesn't cover the fact that their heaviest armor has 45 melee defense while helmets a very smol 15 melee, they also can't wear gloves or combat boots that give marines some extra melee defense points too. Their snowflake armor does not have good amount of space for jobs like medics or engineers where they have jeager storage modules. 30% acid damage intake increase is fine however the healing time... So freaking long for both brute and burn damage and since they can't process pills, no passive regen either, also the only way you won't have to make trips back and forth to fob to refill your welder is if you take like 4 of them with you wasting even more valuable space or play as engineer but then you have less space for materials and other engi gadgets.

I would suggest buffing their passive melee resist to 35% and their helmet armor values from 15 to like 30, decreased time to heal each limb with welder and/or cable would be also good.

File: code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm [hunk 2]
```diff
@@ -476,7 +476,9 @@ GLOBAL_LIST_INIT(wraith_banish_very_short_duration_list, typecacheof(list(
 			if(ishuman(pulled_target))
 				var/mob/living/carbon/human/H = pulled_target
 				if(H.stat == UNCONSCIOUS) //Apply critdrag damage as if they were quickly pulled the same distance
-					H.adjustOxyLoss(HUMAN_CRITDRAG_OXYLOSS * get_dist(H.loc, T))
+					var/critdamage = HUMAN_CRITDRAG_OXYLOSS * get_dist(H.loc, T)
+					if(!H.adjustOxyLoss(critdamage))
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm:480]: why are you not using a species flag here?

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm:480]: General solution if say we add a nooxy implant/chem or such it stops bodydrag from being an issue while still punishing

File: code/datums/jobs/job/job.dm [hunk 3]
```diff
@@ -71,7 +71,7 @@ GLOBAL_PROTECT(exp_specialmap)
 /datum/job/proc/after_spawn(mob/living/L, mob/M, latejoin = FALSE) //do actions on L but send messages to M as the key may not have been transferred_yet
 	if(!ishuman(L))
 		return
-
+	var/mob/living/carbon/human/H = L
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/datums/jobs/job/job.dm:74]: Because ishuman check?

File: code/game/sound.dm [hunk 4]
```diff
@@ -286,6 +286,7 @@ A good representation is: 'byond applies a volume reduction to the sound every X
 			S = pick('sound/voice/alien_queen_command.ogg','sound/voice/alien_queen_command2.ogg','sound/voice/alien_queen_command3.ogg')
 		if("alien_ventpass")
 			S = pick('sound/effects/alien_ventpass1.ogg', 'sound/effects/alien_ventpass2.ogg')
+
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/game/sound.dm:289]: on purpose, this switch is hard to read at times

**Review by BraveMole — APPROVED** [review]

**Pull request merged**